src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -1700,12 +1712,7 @@ JavaThreadStatus java_lang_Thread::get_thread_status(oop java_thread) {
   assert(Threads_lock->owned_by_self() || Thread::current()->is_VM_thread() ||
          JavaThread::current()->thread_state() == _thread_in_vm,
          "Java Thread is not running in vm");
-  oop holder = java_lang_Thread::holder(java_thread);
-  if (holder == nullptr) {
-    return JavaThreadStatus::NEW;  // Java Thread not initialized
-  } else {
-    return java_lang_Thread_FieldHolder::get_thread_status(holder);
-  }
+  GET_FIELDHOLDER_FIELD(java_thread, get_thread_status, JavaThreadStatus::NEW;  /* not initialized */);
```

Comment:
I didn't expect to see a ';' after the 'NEW'. Should that be deleted?
Should it have caused a compile error?

Reply:
Well spotted. It isn't wanted but doesn't cause an error - not sure why not as it is an unreachable empty statement. Anyway it is fixed.